### DISCUSSION TITLE
matchUpdateTypes not working, always pulling in also major updates

### DISCUSSION THREAD

**Discussion by marcelser** [renovatebot/renovate#31865] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

None

### Please tell us more about your question or problem

We're about to introduce renovatebot but in the onboarding PR renovate tells us that it will create also MRs for major changes. We tried to limit that with matchUpdatTypes setting but it doesn't work. When I run it with logtype debug I see in the config that it uses matchUpdateTypes but it doesn't work.

The 2nd packageRule is atm just for debugging and will be removed, it's a copy of the first but it just it runs very fast in combination with an autodiscovery filter for a specific repo. Apart from the schedule and the matchBaseBranch it's identical to the first (which should in the end become the one we use weekly). In addition it should create PRs for vulnerability issues found in OSV no matter which updateType it is.

But I just can't figure out whats going and why it's still creating MRs for major changes. In onboarding MR I see it will create MRs for the minor&patch change and one for the major for some components which is exactly what the docs are listing if you don't use matchUpdateTypes.

Here's my config:
```
module.exports = {
  "onboardingConfig": {
    "configMigration": true,
    "labels": ["renovate"],
    "extends": ["config:recommended"],
    "dependencyDashboardOSVVulnerabilitySummary": "all",
    "osvVulnerabilityAlerts": true,
    "baseBranches": ["AD-25403"],
    "vulnerabilityAlerts": {
      "enabled": true,
      "addLabels": ["security"],
    },
      "packageRules": [
        {
          "enabled": true,
          "matchPackageNames": ["*"],
          "matchUpdateTypes": ["minor", "patch"],
          "automerge": false,
          "timezone": "Europe/Zurich",
          "schedule": ["on saturday"],
        },
        {
          "enabled": true,
          "matchBaseBranches": ["AD-25403"],
          "automerge": false,
          "timezone": "Europe/Zurich",
          "matchUpdateTypes": ["minor", "patch"],
          "schedule": ["* * * * *"],        
        }
      ]
  },
};
```

I also tried  moving the packageRules outside of the onboarding config but then it complains that I have to define the baseBranches for the matchBaseBranches section which is stupid. They are not inside the onboardingconfig and should still work.

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Using RE2 regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in ./main-config.js
DEBUG: Converting GITHUB_COM_TOKEN into a global host rule
DEBUG: File config
       "config": {
         "onboardingConfig": {
           "configMigration": true,
           "labels": ["renovate"],
           "extends": ["config:recommended"],
           "dependencyDashboardOSVVulnerabilitySummary": "all",
           "osvVulnerabilityAlerts": true,
           "baseBranches": ["AD-25403"],
           "vulnerabilityAlerts": {"enabled": true, "addLabels": ["security"]},
           "packageRules": [
             {
               "enabled": true,
               "matchPackageNames": ["*"],
               "matchUpdateTypes": ["minor", "patch"],
               "automerge": false,
               "timezone": "Europe/Zurich",
               "schedule": ["on saturday"]
             },
             {
               "enabled": true,
               "matchBaseBranches": ["AD-25403"],
               "automerge": false,
               "timezone": "Europe/Zurich",
               "matchUpdateTypes": ["minor", "patch"],
               "schedule": ["* * * * *"]
             }
           ]
         }
       }
DEBUG: CLI config
       "config": {
         "platform": "gitlab",
         "autodiscover": true,
         "autodiscoverFilter": "dauntless/umantis"
       }
DEBUG: Env config
       "config": {
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ],
         "repositoryCache": "enabled",
         "baseDir": "/builds/dauntless/platform/renovate-runner/renovate",
         "optimizeForDisabled": true,
         "platform": "gitlab",
         "endpoint": "https://*************/api/v4",
         "token": "***********"
       }
DEBUG: Combined config
       "config": {
         "onboardingConfig": {
           "configMigration": true,
           "labels": ["renovate"],
           "extends": ["config:recommended"],
           "dependencyDashboardOSVVulnerabilitySummary": "all",
           "osvVulnerabilityAlerts": true,
           "baseBranches": ["AD-25403"],
           "vulnerabilityAlerts": {"enabled": true, "addLabels": ["security"]},
           "packageRules": [
             {
               "enabled": true,
               "matchPackageNames": ["*"],
               "matchUpdateTypes": ["minor", "patch"],
               "automerge": false,
               "timezone": "Europe/Zurich",
               "schedule": ["on saturday"]
             },
             {
               "enabled": true,
               "matchBaseBranches": ["AD-25403"],
               "automerge": false,
               "timezone": "Europe/Zurich",
               "matchUpdateTypes": ["minor", "patch"],
               "schedule": ["* * * * *"]
             }
           ]
         },
         "hostRules": [
           {"hostType": "github", "matchHost": "github.com", "token": "***********"}
         ],
         "repositoryCache": "enabled",
         "baseDir": "/builds/dauntless/platform/renovate-runner/renovate",
         "optimizeForDisabled": true,
         "platform": "gitlab",
         "endpoint": "https://*************/api/v4",
         "token": "***********",
         "autodiscover": true,
         "autodiscoverFilter": "dauntless/umantis"
       }
DEBUG: Adding trailing slash to endpoint
DEBUG: Found valid git version: 2.46.0
DEBUG: Setting global hostRules
DEBUG: Adding token authentication for github.com (hostType=github) to hostRules
DEBUG: hostRules: authentication already set for *************
DEBUG: Using queue: host=*************, concurrency=16
DEBUG: GitLab version is: 17.4.1-ee
DEBUG: Using platform gitAuthor: renovatebot runner <melser_regs@gmx.net>
DEBUG: Adding token authentication for ************* (hostType=gitlab) to hostRules
DEBUG: Using configured baseDir: /builds/dauntless/platform/renovate-runner/renovate
DEBUG: Using cacheDir: /builds/dauntless/platform/renovate-runner/renovate/cache
DEBUG: Using containerbaseDir: /builds/dauntless/platform/renovate-runner/renovate/cache/containerbase
DEBUG: Initializing Renovate internal cache into /builds/dauntless/platform/renovate-runner/renovate/cache/renovate/renovate-cache-v1
DEBUG: Commits limit = null
DEBUG: Setting global hostRules
DEBUG: Adding token authentication for github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for ************* (hostType=gitlab) to hostRules
DEBUG: validatePresets()
DEBUG: Autodiscovering GitLab repositories
DEBUG: hostRules: applying Bearer authentication for *************
DEBUG: Discovered 95 project(s)
DEBUG: Autodiscovered 95 repositories
DEBUG: Applying autodiscoverFilter
       "autodiscoverFilter": "dauntless/umantis"
 INFO: Autodiscovered repositories
       "length": 1,
       "repositories": ["dauntless/umantis"]
DEBUG: Reinitializing hostRules for repo
DEBUG: Clearing hostRules
DEBUG: Adding token authentication for github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for ************* (hostType=gitlab) to hostRules
 INFO: Repository started (repository=dauntless/umantis)
       "renovateVersion": "38.84.0"
DEBUG: Using localDir: /builds/dauntless/platform/renovate-runner/renovate/repos/gitlab/dauntless/umantis (repository=dauntless/umantis)
DEBUG: PackageFiles.clear() - Package files deleted (repository=dauntless/umantis)
DEBUG: hostRules: applying Bearer authentication for ************* (repository=dauntless/umantis)
DEBUG: Using queue: host=*************, concurrency=16 (repository=dauntless/umantis)
DEBUG: dauntless/umantis default branch = master (repository=dauntless/umantis)
DEBUG: Enabling Git FS (repository=dauntless/umantis)
DEBUG: Using http URL: https://*************/dauntless/umantis.git (repository=dauntless/umantis)
DEBUG: GET https://*************/api/v4/projects/dauntless%2Fumantis/repository/files/renovate.json?ref=HEAD = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=135) (repository=dauntless/umantis)
DEBUG: GitLab API 404 (repository=dauntless/umantis)
       "url": "https://*************/api/v4/projects/dauntless%2Fumantis/repository/files/renovate.json?ref=HEAD"
DEBUG: Repository cache is restored from revision 13 (repository=dauntless/umantis)
DEBUG: Resetting npmrc (repository=dauntless/umantis)
DEBUG: Resetting npmrc (repository=dauntless/umantis)
DEBUG: checkOnboarding() (repository=dauntless/umantis)
DEBUG: isOnboarded() (repository=dauntless/umantis)
DEBUG: findPr(renovate/configure, Configure Renovate, !open) (repository=dauntless/umantis)
DEBUG: findPr(renovate/configure, chore: Configure Renovate, !open) (repository=dauntless/umantis)
DEBUG: Onboarding cache is valid. Repo is not onboarded (repository=dauntless/umantis)
DEBUG: Repo is not onboarded (repository=dauntless/umantis)
DEBUG: getBranchPr(renovate/configure) (repository=dauntless/umantis)
DEBUG: findPr(renovate/configure, undefined, open) (repository=dauntless/umantis)
DEBUG: getPr(5323) (repository=dauntless/umantis)
DEBUG: getMR(5323) (repository=dauntless/umantis)
DEBUG: Onboarding PR already exists (repository=dauntless/umantis)
DEBUG: Checking if onboarding branch needs rebasing (repository=dauntless/umantis)
DEBUG: Checking for a default Renovate preset which can be used. (repository=dauntless/umantis)
DEBUG: Checking for preset: local>dauntless/renovate-config (repository=dauntless/umantis)
DEBUG: GET https://*************/api/v4/projects/dauntless%2Frenovate-config = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=78) (repository=dauntless/umantis)
DEBUG: GitLab API 404 (repository=dauntless/umantis)
       "url": "https://*************/api/v4/projects/dauntless%2Frenovate-config"
DEBUG: Preset file default.json not found in dauntless/renovate-config (repository=dauntless/umantis)
DEBUG: GitLab API 404 (repository=dauntless/umantis)
       "url": "https://*************/api/v4/projects/dauntless%2Frenovate-config"
DEBUG: Preset file renovate.json not found in dauntless/renovate-config (repository=dauntless/umantis)
DEBUG: Checking for preset: local>dauntless/.gitlab:renovate-config (repository=dauntless/umantis)
DEBUG: GET https://*************/api/v4/projects/dauntless%2F.gitlab = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=74) (repository=dauntless/umantis)
DEBUG: GitLab API 404 (repository=dauntless/umantis)
       "url": "https://*************/api/v4/projects/dauntless%2F.gitlab"
DEBUG: Preset file renovate-config.json not found in dauntless/.gitlab (repository=dauntless/umantis)
DEBUG: No default org/owner preset found, so the default onboarding config will be used instead. (repository=dauntless/umantis)
DEBUG: onboarding config (repository=dauntless/umantis)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "configMigration": true,
         "labels": ["renovate"],
         "extends": ["config:recommended"],
         "dependencyDashboardOSVVulnerabilitySummary": "all",
         "osvVulnerabilityAlerts": true,
         "baseBranches": ["AD-25403"],
         "vulnerabilityAlerts": {"enabled": true, "addLabels": ["security"]},
         "packageRules": [
           {
             "enabled": true,
             "matchPackageNames": ["*"],
             "matchUpdateTypes": ["minor", "patch"],
             "automerge": false,
             "timezone": "Europe/Zurich",
             "schedule": ["on saturday"]
           },
           {
             "enabled": true,
             "matchBaseBranches": ["AD-25403"],
             "automerge": false,
             "timezone": "Europe/Zurich",
             "matchUpdateTypes": ["minor", "patch"],
             "schedule": ["* * * * *"]
           }
         ]
       }
DEBUG: No rebase needed (repository=dauntless/umantis)
DEBUG: Skip processing since the onboarding branch is up to date and default branch has not changed (repository=dauntless/umantis)
DEBUG: GitLab API 404 (repository=dauntless/umantis)
       "url": "https://*************/api/v4/projects/dauntless%2Fumantis/repository/files/renovate.json?ref=HEAD"
DEBUG: Existing config file no longer exists (repository=dauntless/umantis)
DEBUG: Found renovate.json config file (repository=dauntless/umantis)
DEBUG: migrateAndValidate() (repository=dauntless/umantis)
DEBUG: No config migration necessary (repository=dauntless/umantis)
DEBUG: Found repo ignorePaths (repository=dauntless/umantis)
       "ignorePaths": [
         "**/node_modules/**",
         "**/bower_components/**",
         "**/vendor/**",
         "**/examples/**",
         "**/__tests__/**",
         "**/test/**",
         "**/tests/**",
         "**/__fixtures__/**"
       ]
DEBUG: No vulnerability alerts found (repository=dauntless/umantis)
DEBUG: detectSemanticCommits() (repository=dauntless/umantis)
DEBUG: semanticCommits: returning "disabled" from cache (repository=dauntless/umantis)
DEBUG: processRepo() (repository=dauntless/umantis)
DEBUG: GitLab API 404 (repository=dauntless/umantis)
       "url": "https://*************/api/v4/projects/dauntless%2Fumantis/repository/files/renovate.json?ref=HEAD"
DEBUG: Existing config file no longer exists (repository=dauntless/umantis)
DEBUG: Found renovate.json config file (repository=dauntless/umantis)
DEBUG: checkConfigMigrationBranch() (repository=dauntless/umantis)
DEBUG: checkConfigMigrationBranch() Config does not need migration (repository=dauntless/umantis)
DEBUG: ensureDependencyDashboard() (repository=dauntless/umantis)
DEBUG: Repo is onboarding - skipping dependency dashboard (repository=dauntless/umantis)
DEBUG: validateReconfigureBranch() (repository=dauntless/umantis)
DEBUG: No reconfigure branch found (repository=dauntless/umantis)
DEBUG: Removing any stale branches (repository=dauntless/umantis)
DEBUG: config.repoIsOnboarded=false (repository=dauntless/umantis)
DEBUG: Branch lists (repository=dauntless/umantis)
       "branchList": ["renovate/configure"],
       "renovateBranches": ["renovate/configure"]
DEBUG: remainingBranches= (repository=dauntless/umantis)
DEBUG: No branches to clean up (repository=dauntless/umantis)
DEBUG: ensureIssueClosing() (repository=dauntless/umantis)
DEBUG: ensureIssueClosing() (repository=dauntless/umantis)
DEBUG: PackageFiles.clear() - Package files deleted (repository=dauntless/umantis)
DEBUG: Branch summary (repository=dauntless/umantis)

```

</details>
<details>
<summary>MRs</summary>

.... at the end of the logs you see a lot of branches it will create with type 'major'

      {
           "branchName": "renovate/start-server-and-test-2.x",
           "prNo": null,
           "prTitle": "Update dependency start-server-and-test to v2",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "start-server-and-test",
               "displayPending": "",
               "fixedVersion": "1.15.4",
               "currentVersion": "1.15.4",
               "currentValue": "=1.15.4",
               "newValue": "=2.0.8",
               "newVersion": "2.0.8",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "start-server-and-test"
             }
           ]
         },
         {
           "branchName": "renovate/supertest-7.x",
           "prNo": null,
           "prTitle": "Update dependency supertest to v7",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "supertest",
               "displayPending": "",
               "fixedVersion": "6.3.4",
               "currentVersion": "6.3.4",
               "currentValue": "^6.3.4",
               "newValue": "^7.0.0",
               "newVersion": "7.0.0",
               "packageFile": "testing/api-chai-mocha/package.json",
               "updateType": "major",
               "packageName": "supertest"
             }
           ]
         },
         {
           "branchName": "renovate/terser-5.x",
           "prNo": null,
           "prTitle": "Update dependency terser to v5",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "terser",
               "displayPending": "",
               "fixedVersion": "4.8.1",
               "currentVersion": "4.8.1",
               "currentValue": "^4.6.3",
               "newValue": "^5.0.0",
               "newVersion": "5.34.1",
               "packageFile": "view/package.json",
               "updateType": "major",
               "packageName": "terser"
             }
           ]
         },
         {
           "branchName": "renovate/terser-webpack-plugin-5.x",
           "prNo": null,
           "prTitle": "Update dependency terser-webpack-plugin to v5",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "terser-webpack-plugin",
               "displayPending": "",
               "fixedVersion": "2.3.8",
               "currentVersion": "2.3.8",
               "currentValue": "^2.3.4",
               "newValue": "^5.0.0",
               "newVersion": "5.3.10",
               "packageFile": "view/package.json",
               "updateType": "major",
               "packageName": "terser-webpack-plugin"
             },
             {
               "datasource": "npm",
               "depName": "terser-webpack-plugin",
               "displayPending": "",
               "currentVersion": "4.2.3",
               "currentValue": "^4.2.3",
               "newValue": "^5.0.0",
               "newVersion": "5.3.10",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "terser-webpack-plugin"
             }
           ]
         },
         {
           "branchName": "renovate/webpack-5.x",
           "prNo": null,
           "prTitle": "Update dependency webpack to v5",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "webpack",
               "displayPending": "",
               "fixedVersion": "4.47.0",
               "currentVersion": "4.47.0",
               "currentValue": "^4.47.0",
               "newValue": "^5.0.0",
               "newVersion": "5.95.0",
               "packageFile": "view/package.json",
               "updateType": "major",
               "packageName": "webpack"
             },
             {
               "datasource": "npm",
               "depName": "webpack",
               "displayPending": "",
               "currentVersion": "4.47.0",
               "currentValue": "^4.46.0",
               "newValue": "^5.0.0",
               "newVersion": "5.95.0",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "webpack"
             }
           ]
         },
         {
           "branchName": "renovate/webpack-cli-5.x",
           "prNo": null,
           "prTitle": "Update dependency webpack-cli to v5",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "webpack-cli",
               "displayPending": "",
               "fixedVersion": "3.3.12",
               "currentVersion": "3.3.12",
               "currentValue": "^3.3.10",
               "newValue": "^5.0.0",
               "newVersion": "5.1.4",
               "packageFile": "view/package.json",
               "updateType": "major",
               "packageName": "webpack-cli"
             },
             {
               "datasource": "npm",
               "depName": "webpack-cli",
               "displayPending": "",
               "currentVersion": "4.10.0",
               "currentValue": "^4.7.2",
               "newValue": "^5.0.0",
               "newVersion": "5.1.4",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "webpack-cli"
             }
           ]
         },
         {
           "branchName": "renovate/webpack-dev-server-5.x",
           "prNo": null,
           "prTitle": "Update dependency webpack-dev-server to v5",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "webpack-dev-server",
               "displayPending": "",
               "currentVersion": "3.11.3",
               "currentValue": "^3.11.2",
               "newValue": "^5.0.0",
               "newVersion": "5.1.0",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "webpack-dev-server"
             }
           ]
         },
         {
           "branchName": "renovate/major-jest-monorepo",
           "prNo": null,
           "prTitle": "Update jest monorepo to v29 (major)",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "@jest/globals",
               "displayPending": "",
               "fixedVersion": "27.5.0",
               "currentVersion": "27.5.0",
               "currentValue": "=27.5.0",
               "newValue": "=29.7.0",
               "newVersion": "29.7.0",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "@jest/globals"
             },
             {
               "datasource": "npm",
               "depName": "expect",
               "displayPending": "",
               "fixedVersion": "27.5.0",
               "currentVersion": "27.5.0",
               "currentValue": "=27.5.0",
               "newValue": "=29.7.0",
               "newVersion": "29.7.0",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "expect"
             },
             {
               "datasource": "npm",
               "depName": "jest",
               "displayPending": "",
               "fixedVersion": "27.5.0",
               "currentVersion": "27.5.0",
               "currentValue": "=27.5.0",
               "newValue": "=29.7.0",
               "newVersion": "29.7.0",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "jest"
             }
           ]
         },
         {
           "branchName": "renovate/io.freefair.lombok-8.x",
           "prNo": null,
           "prTitle": "Update plugin io.freefair.lombok to v8",
           "upgrades": [
             {
               "datasource": "maven",
               "depName": "io.freefair.lombok",
               "displayPending": "",
               "fixedVersion": "6.6.3",
               "currentVersion": "6.6.3",
               "currentValue": "6.6.3",
               "newValue": "8.10.2",
               "newVersion": "8.10.2",
               "packageFile": "testing/api-integration-tests/settings.gradle",
               "updateType": "major",
               "packageName": "io.freefair.lombok:io.freefair.lombok.gradle.plugin"
             }
           ]
         },
         {
           "branchName": "renovate/major-react-monorepo",
           "prNo": null,
           "prTitle": "Update react monorepo to v18 (major)",
           "upgrades": [
             {
               "datasource": "npm",
               "depName": "react",
               "displayPending": "",
               "fixedVersion": "17.0.2",
               "currentVersion": "17.0.2",
               "currentValue": "=17.0.2",
               "newValue": "=18.3.1",
               "newVersion": "18.3.1",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "react"
             },
             {
               "datasource": "npm",
               "depName": "react-dom",
               "displayPending": "",
               "fixedVersion": "17.0.2",
               "currentVersion": "17.0.2",
               "currentValue": "=17.0.2",
               "newValue": "=18.3.1",
               "newVersion": "18.3.1",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "react-dom"
             },
             {
               "datasource": "npm",
               "depName": "react-test-renderer",
               "displayPending": "",
               "fixedVersion": "17.0.2",
               "currentVersion": "17.0.2",
               "currentValue": "=17.0.2",
               "newValue": "=18.3.1",
               "newVersion": "18.3.1",
               "packageFile": "API/StaticAssets/package.json",
               "updateType": "major",
               "packageName": "react-test-renderer"
             }
           ]
         },
         {
           "branchName": "renovate/major-spring-boot",
           "prNo": null,
           "prTitle": "Update spring boot to v3 (major)",
           "upgrades": [
             {
               "datasource": "maven",
               "depName": "org.springframework.boot:spring-boot-starter-test",
               "displayPending": "",
               "fixedVersion": "2.7.5",
               "currentVersion": "2.7.5",
               "currentValue": "2.7.5",
               "newValue": "3.3.4",
               "newVersion": "3.3.4",
               "packageFile": "testing/ui-test/selenium/pom.xml",
               "updateType": "major",
               "packageName": "org.springframework.boot:spring-boot-starter-test"
             },
             {
               "datasource": "maven",
               "depName": "org.springframework.boot:spring-boot-starter-parent",
               "displayPending": "",
               "fixedVersion": "2.7.18",
               "currentVersion": "2.7.18",
               "currentValue": "2.7.18",
               "newValue": "3.3.4",
               "newVersion": "3.3.4",
               "packageFile": "testing/ui-test/selenium/pom.xml",
               "updateType": "major",
               "packageName": "org.springframework.boot:spring-boot-starter-parent"
             },
             {
               "datasource": "maven",
               "depName": "org.springframework.boot:spring-boot-maven-plugin",
               "displayPending": "",
               "fixedVersion": "2.7.5",
               "currentVersion": "2.7.5",
               "currentValue": "2.7.5",
               "newValue": "3.3.4",
               "newVersion": "3.3.4",
               "packageFile": "testing/ui-test/selenium/ci-tests/pom.xml",
               "updateType": "major",
               "packageName": "org.springframework.boot:spring-boot-maven-plugin"
             },
             {
               "datasource": "maven",
               "depName": "org.springframework.boot",
               "displayPending": "",
               "fixedVersion": "2.7.18",
               "currentVersion": "2.7.18",
               "currentValue": "2.7.18",
               "newValue": "3.3.4",
               "newVersion": "3.3.4",
               "packageFile": "testing/api-integration-tests/settings.gradle",
               "updateType": "major",
               "packageName": "org.springframework.boot:org.springframework.boot.gradle.plugin"
             }

</summary>
</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
You haven't disabled major updates in any of the package rule. Try that

        {
          "enabled": false,
          "matchUpdateTypes": ["major"]
        },

**Reply by marcelser** [reply]
That did the trick, thank you
